[CI] FileSettingsRoleMappingUpgradeIT testRoleMappingsAppliedOnUpgrade {upgradedNodes=3} failing

**Build Scans:**
- [elasticsearch-periodic #4543 / 8.8.2_bwc](https://gradle-enterprise.elastic.co/s/6o6v3sf4co7c6)
- [elasticsearch-periodic #4543 / 8.7.1_bwc](https://gradle-enterprise.elastic.co/s/dnscy5tvnnwbi)

**Reproduction Line:**
```
./gradlew ":qa:rolling-upgrade:v8.7.1#bwcTest" -Dtests.class="org.elasticsearch.upgrades.FileSettingsRoleMappingUpgradeIT" -Dtests.method="testRoleMappingsAppliedOnUpgrade {upgradedNodes=3}" -Dtests.seed=F6ADB9CD20C85AAC -Dtests.bwc=true -Dtests.locale=nb -Dtests.timezone=Pacific/Port_Moresby -Druntime.java=22
```

**Applicable branches:**
8.16

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRoleMappingsAppliedOnUpgrade%20%7BupgradedNodes%3D3%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.upgrades.FileSettingsRoleMappingUpgradeIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is an instance of java.util.Map
     but: <[{metadata={_es_reserved_role_mapping_name=everyone_kibana}, roles=[kibana_user], rules={field={username=*}}, enabled=true}]> is a java.util.ArrayList
```

**Issue Reasons:**
- [8.16] 2 consecutive failures in test testRoleMappingsAppliedOnUpgrade {upgradedNodes=3}
- [8.16] 2 failures in test testRoleMappingsAppliedOnUpgrade {upgradedNodes=3} (5.7% fail rate in 35 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FileSettingsRoleMappingUpgradeIT.java
+++ b/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FileSettingsRoleMappingUpgradeIT.java
@@ -25,12 +25,10 @@ import org.junit.rules.TestRule;
 
 import java.io.IOException;
 import java.util.List;
-import java.util.Map;
 import java.util.function.Supplier;
 
-import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
@@ -104,15 +102,17 @@ public class FileSettingsRoleMappingUpgradeIT extends ParameterizedRollingUpgrad
             // the nodes have all been upgraded. Check they re-processed the role mappings in the settings file on
             // upgrade
             Request clusterStateRequest = new Request("GET", "/_cluster/state/metadata");
-            List<Object> roleMappings = new XContentTestUtils.JsonMapView(entityAsMap(client().performRequest(clusterStateRequest))).get(
-                "metadata.role_mappings.role_mappings"
+            List<Object> clusterStateRoleMappings = new XContentTestUtils.JsonMapView(
+                entityAsMap(client().performRequest(clusterStateRequest))
+            ).get("metadata.role_mappings.role_mappings");
+            assertThat(clusterStateRoleMappings, is(not(nullValue())));
+            assertThat(clusterStateRoleMappings.size(), equalTo(1));
+
+            assertThat(
+                entityAsMap(client().performRequest(new Request("GET", "/_security/role_mapping"))).keySet(),
+                // TODO remove `everyone_kibana` once the clean up migration is in
+                containsInAnyOrder("everyone_kibana", "everyone_kibana-read-only-operator-mapping")
             );
-            assertThat(roleMappings, is(not(nullValue())));
-            assertThat(roleMappings.size(), equalTo(1));
-            assertThat(roleMappings, is(instanceOf(Map.class)));
-            @SuppressWarnings("unchecked")
-            Map<String, Object> roleMapping = (Map<String, Object>) roleMappings;
-            assertThat(roleMapping.keySet(), contains("everyone_kibana-read-only-operator-mapping"));
         }
     }
 }

--- a/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FileSettingsRoleMappingUpgradeIT.java
+++ b/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FileSettingsRoleMappingUpgradeIT.java
@@ -27,8 +27,8 @@ import java.io.IOException;
 import java.util.List;
 import java.util.function.Supplier;
 
-import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
@@ -110,8 +110,8 @@ public class FileSettingsRoleMappingUpgradeIT extends ParameterizedRollingUpgrad
 
             assertThat(
                 entityAsMap(client().performRequest(new Request("GET", "/_security/role_mapping"))).keySet(),
-                // TODO remove `everyone_kibana` once the clean up migration is in
-                containsInAnyOrder("everyone_kibana", "everyone_kibana-read-only-operator-mapping")
+                // TODO change this to `contains` once the clean-up migration work is merged
+                hasItem("everyone_kibana-read-only-operator-mapping")
             );
         }
     }

